### PR TITLE
Viewer clock improvements

### PR DESCRIPTION
**Pull request by mramato**
In master, it's impossible specify the `clockViewModel` instance to use for any child widgets that need it.  Viewer also doesn't expose the `ClockViewModel` and you have to dig into `animation.viewModel.clockViewModel` to get to it.  This makes it hard to write custom widgets that need to share the same `ClockViewModel` and are not directly tied to the lifetime of the 
 clockViewModel` created by Viewer (which gets destroyed when the Viewer is destroyed).

This change deprecates the Viewer `clock` constructor option and replaces it with `clockViewModel`.  If a custom `clockViewModel` is specified, it is passed through to any child widgets that need it and it is not destroyed on destruction.

We didn't have tests for the old functionality, but I added a test for the new parameter.

### CODE REVIEW

File: Source/Widgets/Viewer/Viewer.js [hunk 1]
```diff
@@ -401,11 +405,27 @@ Either specify options.terrainProvider instead or set options.baseLayerPicker to
...
+        }
+
+        var clock;
+        var clockViewModel;
+        var destroyClockViewModel = false;
+        if (defined(options.clockViewModel)) {
+            clockViewModel = options.clockViewModel;
+            clock = clockViewModel.clock;
+        } else {
+            clock = new Clock();
```
Comment (hpinkos) [inline Source/Widgets/Viewer/Viewer.js]: Shouldn't this be `defaultValue(options.clock, new Clock())` to maintain the existing behavior until `options.clock` is removed?

**Comment by hpinkos** [comment]
Why can't we just get the `ClockViewModel` that's created by the viewer?  Because you need it to stay around after the viewer is destroyed?

**Comment by emackey** [comment]
Also what happens if you want multiple viewers to share a single clock?  Is that still supported?  I'm using `options.clock` in at least one production app.

File: Source/Widgets/Viewer/Viewer.js (hunk 1 above)
Comment (mramato) [inline Source/Widgets/Viewer/Viewer.js]: Yep, good catch.  Just pushed a fix.

**Comment by mramato** [comment]
> Also what happens if you want multiple viewers to share a single clock? Is that still supported? I'm using options.clock in at least one production app.

Sure, you get the same exact effect by making them share the same `clockViewModel`.  In fact, I wouldn't be surprised if there were potential bugs in the old behavior because you ended up with two `clockViewModel` instances controlling the same clock.  (plus, if anyone really needs the exact old behavior you can still create two `clockViewModel` instances that have the same clock and pass them too each Viewer).

**Comment by mramato** [comment]
> Why can't we just get the ClockViewModel that's created by the viewer? Because you need it to stay around after the viewer is destroyed?

Yes, but also because in my use case, the `ClockViewModel` needs to exist before the Animation widget is even constructed.  Keep in mind that `AnimationViewModel` already requires an existing `ClockViewModel` instance, it just that previously the `Viewer` was always the one creating it.

**Comment by mramato** [comment]
This is ready unless either of you still have some reservations or more questions/comments.

**Comment by emackey** [comment]
I'm fine with this.  @hpinkos please merge when ready.

**Comment by hpinkos** [comment]
:+1: looks good to me!

**Pull request merged**